site/releases/2024-may-22/release-notes.qmd

```diff
@@ -0,0 +1,635 @@
+
+<!---
+PR #76: [SC 4208] Define and implement required text data validation tests
+URL: https://github.com/validmind/developer-framework/pull/76
+Labels: enhancement
+--->
+### New text data validation tests
+
+The following tests for text data validation have been added:
+
+- `validmind.data_validation.nlp.LanguageDetection`
+- `validmind.data_validation.nlp.Toxicity`
+- `validmind.data_validation.nlp.PolarityAndSubjectivity`
+- `validmind.data_validation.nlp.Sentiment`
+
+<!--- NR TO DO This notebook is not in main
+To test run notebook: `rag_langchain_demo.ipynb` --->
+
+<!---
+PR #44: John6797/sc 3882/metric decorator should work with test providers
+URL: https://github.com/validmind/developer-framework/pull/44
+Labels: enhancement
+--->
+### Use metric decorators with test providers
+
+Continuing our efforts to simplify the process for getting your custom code working in ValidMind, we now support functional metrics for internal tests and test providers. Functional metrics are tests that can be defined as pure functions without any boilerplate.
+
+::: {.callout-tip appearance="minimal" title="Try it on JupyterHub"}
+[Integrate external test providers](https://jupyterhub.validmind.ai/hub/user-redirect/lab/tree/code_samples/custom_tests/integrate_external_test_providers.ipynb).
+:::
+
+<!---
+PR #48: John6797/sc 3881/support for metadata in new metric decorator
+URL: https://github.com/validmind/developer-framework/pull/48
+Labels: enhancement
+--->
+### Support for metadata in new metric decorator
+
+We added new decorators to support task type and tag metadata in functional metrics.
+
+**Usage example:**
+
+```python
+from sklearn.metrics import accuracy_score
+
+from validmind import tags, tasks
+
+
+@tasks("classification")
+@tags("classification", "sklearn", "accuracy")
+def Accuracy(dataset, model):
+    """Calculates the accuracy of a model"""
+    return accuracy_score(dataset.y, dataset.y_pred(model))
+
+
+# the above decorator is syntactic sugar for the following:
+Accuracy.__tags__ = ["classification"]
+Accuracy.__tasks__ = ["classification", "sklearn", "accuracy"]
+```
+
+<!---
+PR #43: [SC-3423] Review credit scorecard notebook
+URL: https://github.com/validmind/developer-framework/pull/43
+Labels: enhancement
+--->
+### Assign prediction probabilities
+
+We added support for assigning `prediction_probabilities` to `assign_predictions`. This support enables you to:
+
+- Assign prediction values and probabilities that have been computed outside ValidMind.
+- Incorporate prediction values and probabilities from datasets that already have prediction columns.
+- Automate the assignment of prediction values and probabilities within VM.
+
+::: {.callout-tip appearance="minimal" title="Try it"}
+[Assign Prediction Values and Probabilities to ValidMind Datasets](https://github.com/validmind/developer-framework/blob/main/notebooks/code_sharing/credit_risk/assign_prediction_probabilities.ipynb)
```

Comment:
Is this meant to be a JupyterHub link? Also, this notebook still seems to be in our internal `code_sharing` folder.

Reply:
This is the only way to get to the notebook, as this folder isn't included in JupyterHub or our docs site. At the same this, it's open source, so as a first-time experiment I actually just linked ... GitHub.